[NONE] updating changelog (#6824)
prometheus-operator/prometheus-operator: CHANGELOG.md

## 0.76.0 / 2025-08-08

> [CHANGE] Change the `secrets` field of Prometheus and PrometheusAgent CRDs from `atomic` to `set`. This allows different managers to manage entries separately. #6762

Reviewer: As a user who has no idea about developments, I feel this entry is still confusing. What is atomic? what is set? Why should they care? `This allows different managers to manage entries separately`, what entries 😅?

It's categorized as [CHANGED], so what behavior is being changed here?

I'll try to do some investigation, because I also don't understand completely what is changing here 😬

Author: I tend to agree with you, this is confuse and after reading this, I don't think it's a change will impact user and should be on the change log, I'll remove it.

[FEATURE] Add TLS and Proxy settings to OAuth2 configuration for Prometheus and PrometheusAgent CRDs. #6735
[FEATURE] Add support for OAuth2 in the ScrapeConfig CRD. #6814
[FEATURE] Add scale subresource to the Alertmanger CRD, allowing Horizontal Pod Autoscalers(HPAs) to be used. #6728
[FEATURE] Add Scaleway service discovery to the ScrapeConfig CRD. #6711

> [ENHANCEMENT] Make the `namespace` field optional in the Alertmanager endpoints configuration of the Prometheus CRD. #6338

Reviewer: If namespace is now optional, what is the default behavior?

## 0.76.0 / 2025-08-08

[CHANGE] Change the `secrets` field of Prometheus and PrometheusAgent CRDs from `atomic` to `set`. This allows different managers to manage entries separately. #6762
[FEATURE] Add TLS and Proxy settings to OAuth2 configuration for Prometheus and PrometheusAgent CRDs. #6735
[FEATURE] Add support for OAuth2 in the ScrapeConfig CRD. #6814

> [FEATURE] Add scale subresource to the Alertmanger CRD. #6728

Reviewer: ```suggestion
[FEATURE] Add scale subresource to the Alertmanger CRD, allowing Horizontal Pod Autoscalers(HPAs) to be used. #6728
```

We made a big announcement in the changelog when we did it for Prometheus, I think this helps users understand the importance of the scale subresource :)

[FEATURE] Add Scaleway service discovery to the ScrapeConfig CRD. #6711
[ENHANCEMENT] Make the `namespace` field optional in the Alertmanager endpoints configuration of the Prometheus CRD. #6338
[ENHANCEMENT] Add support to configure the TLS version for Prometheus, PrometheusAgent and Alertmanager CRDs. #6736

> [ENHANCEMENT] optimize `TLSConfig` to `addSafeTLSToYaml`. #6726

Reviewer: ```suggestion
```

This doesn't look like a user facing change, does it make sense to remove it?
Remember that we usually don't review PR descriptions before merging them, so the release shepherd needs to use judgment when assembling the changelog 😬

> [ENHANCEMENT] Refactor Kubernetes SD of ScrapeConfig by new adding validations. #6678

Reviewer: ```suggestion
[ENHANCEMENT] Add API-level validations to Kubernetes SD in the ScrapeConfig CRD. #6678
```

Let's remove whatever is not interesting to the final user :) (they don't care about refactorings)

[FEATURE] Add scale subresource to the Alertmanger CRD, allowing Horizontal Pod Autoscalers(HPAs) to be used. #6728
[FEATURE] Add Scaleway service discovery to the ScrapeConfig CRD. #6711
[ENHANCEMENT] Make the `namespace` field optional in the Alertmanager endpoints configuration of the Prometheus CRD, if not defined it will use the `default` namespace. #6338
[ENHANCEMENT] Add support to configure the TLS version for Prometheus, PrometheusAgent and Alertmanager CRDs. #6736
[ENHANCEMENT] Add API-level validations to Kubernetes SD in the ScrapeConfig CRD. #6678
[FEATURE] Add `serviceDiscoveryRole` field to the Prometheus and PrometheusAgent CRDs to select between Endpoints (default) and EndpointSlice for discovering scrape and alerting targets. #6672
[ENHANCEMENT] Add ServiceDiscoveryRole configuration for use of Endpoints vs EndpointSlice. #6672
[ENHANCEMENT] Add `secret-label-selector` argument to the operator to filter the Secrets being watched. #6731
[ENHANCEMENT] Add secret label selector to filter secrets to watch. #6731

> [CHANGE] Add listType:=set to Prometheus secrets field. This allows for having a different manager for each secret. #6762

Reviewer: Hey, sorry if I sounded like I wanted to remove this line from the changelog in the previous review. What I wanted to say is that we need to find a user-friendly log entry.

Looking at [Simon's comment](https://github.com/prometheus-operator/prometheus-operator/pull/6762#pullrequestreview-2191483657), this explains better what the change is, instead of saying `change from atomic to set`.

Could we re-add the entry with the [CHANGE] prefix, but then do one of the two:
1. Improve the wording so it's more user-friendly... or
2. Add a link to Simon's comment, so interested users can read the user-friendly message

Author: Wdyt about the new Change?

> [FEATURE] Add `serviceDiscoveryRole` field to the Prometheus and PrometheusAgent CRDs to select between Endpoints (default) and EndpointSlice for discovering scrape and alerting targets. #6672[ENHANCEMENT] Add ServiceDiscoveryRole configuration for use of Endpoints vs EndpointSlice. #6672

Reviewer: This looks confusing 🤔

Author: Ups

Reviewer: I'm unresolving this comment because we still have 2 entries in the same line? Could you fix that?

> [ENHANCEMENT] Add `secret-label-selector` argument to the operator to filter the Secrets being watched. #6731[ENHANCEMENT] Add secret label selector to filter secrets to watch. #6731

Reviewer: same thing here :O

Reviewer: Same thing here
